scripts/pathfinders_validation.py

```diff
@@ -0,0 +1,439 @@
+from collections import defaultdict
+from datetime import datetime
+from pathlib import Path
+
+import pandas as pd
+import pandera as pa
+
+import tables as ta
+from tables.exceptions import TableExtractError
+
+resources = Path(__file__).parent / "resources"
+
+
+class PFRegex:
+    BASIC_EMAIL = r"^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$"  # will catch most common errors
+    BASIC_TELEPHONE = (
+        r"^(((\+44\s?\d{4}|\(?0\d{4}\)?)\s?\d{3}\s?\d{3})|((\+44\s?\d{3}|\(?0\d{3}\)?)\s?\d{3}"
+        r"\s?\d{4})|((\+44\s?\d{2}|\(?0\d{2}\)?)\s?\d{4}\s?\d{4}))(\s?\#(\d{4}|\d{3}))?$"
+    )
+
+
+class PFEnums:
+    RAGS = ["Green", "Amber/Green", "Amber", "Amber/Red", "Red"]
+    INTERVENTION_THEMES = [
+        "Enhancing sub-regional and regional connectivity",
+        "Unlocking and enabling industrial, commercial, and residential development",
+        "Unlocking and enabling industrial and commercial development",
+        "Strengthening the visitor and local service economy",
+        "Improving the quality of life of residents",
+        "Employment & Education",
+    ]
+    RISK_SCORES = ["1 - very low", "2 - low", "3 - medium", "4 - high", "5 - very high"]
+    FORECAST_ACTUAL = ["Forecast", "Actual"]
+
+
+def get_pathfinders_table_schemas():
+    messages = {
+        "isin": "You’ve entered your own content, instead of selecting from the dropdown list provided. "
+        "Select an option from the dropdown list.",
+        "not_nullable": "The cell is blank but is required.",
+        "field_uniqueness": "You entered duplicate data. Remove or replace the duplicate data.",
+        "multiple_fields_uniqueness": "You entered duplicate data. Remove or replace the duplicate data.",
+        "coerce_dtype": "You entered text instead of a number. Remove any units of measurement and only use numbers, "
+        "for example, 9.",
+    }
+    return {
+        "Financial Completion Date": ta.TableSchema(
+            id_tag="PF-TABLE-ID-01",
+            worksheet_name="Admin",
+            # TODO: are section names needed going forwards? PF doesn't seem to use them - is tab + cell is enough?
+            section="Admin",
+            columns={
+                "Pathfinder Financial Completion Date": pa.Column(datetime),
+            },
+            row_idxs_to_drop=[0],
+            messages={**messages, "coerce_dtype": "You entered text instead of a date."},
```

Comment:
This had crossed my mind as well. I think we could look to do this in future work by possibly extending `Column` and `Check` classes to expect messages. Although, I also see that it could be a bit more laborious as you'd have to define the messages for each `Column` and `Check` each time, whereas here we can just pass a dictionary of shared default values. Perhaps we could include a similar mechanism into a re-worked design though.